rnnalyse/activations/activation_reader.py

```diff
@@ -22,34 +23,154 @@ class ActivationReader:
     Attributes
     ----------
     activations_dir : str
-    labels : np.ndarray
-        Numpy array containing the extracted labels
-    data_len : int
-        Number of extracted activations
+    label_path : str
+    activations : Optional[np.ndarray]
+        Numpy array of activations that are currently read into ram.
+    _labels : Optional[np.ndarray]
+        Numpy array containing the extracted labels. Accessed by the
+        property self.labels.
+    _data_len : int
+        Number of extracted activations. Accessed by the property
+        self.data_len.
+    _activation_ranges : Optional[ActivationRanges]
+        Dictionary mapping sentence keys to their respective location
+        in the .activations array.
     """
+
     def __init__(self,
                  activations_dir: str,
                  label_path: Optional[str] = None) -> None:
 
         self.activations_dir = trim(activations_dir)
 
-        self.labels = self._read_labels(label_path)
-        self.data_len = len(self.labels)
-
-    def _read_labels(self, label_path: Optional[str]) -> np.ndarray:
         if label_path is None:
             label_path = f'{self.activations_dir}/labels.pickle'
+        self.label_path = label_path
+
+        self._activations = None
+        self._labels: Optional[np.ndarray] = None
+        self._data_len: int = -1
+        self._activation_ranges: Optional[ActivationRanges] = None
+
+        self.activations = None
+
+    # TODO: Add examples here or somewhere else
```

Comment:
I think indeed putting some examples into the docstring of this function or the class would be very helpful